packages/web-app-files/tests/unit/components/AppBar/SelectedResources/BatchActions.spec.js

```diff
@@ -1,134 +1,264 @@
-      })
-
-      it('should display the action buttons', () => {
-        const actionButtons = wrapper.findAll(elSelector.ocButton)
-
-        expect(actionButtons.length).toEqual(3)
-        expect(copyButton.text()).toEqual('Copy')
-        expect(moveButton.text()).toEqual('Move')
-        expect(deleteButton.text()).toEqual('Delete')
-      })
 
-      it('should call "triggerLocationPicker" when copy button is clicked', async () => {
-        await copyButton.vm.$emit('click')
-
-        expect(spyTriggerLocationPicker).toHaveBeenCalledWith('copy')
-      })
-
-      it('should call "triggerLocationPicker" when move button is clicked', async () => {
-        await moveButton.vm.$emit('click')
+        if (page === 'files-shared-with-me') {
+          let wrapper
+          let actionButtons
+          let acceptButton
+          let declineButton
+
+          const spyTriggerShareActions = jest
+            .spyOn(BatchActions.methods, 'triggerShareActions')
+            .mockImplementation()
+          const spyAcceptShares = jest.spyOn(BatchActions.methods, 'acceptShares')
+          const spyDeclineShares = jest.spyOn(BatchActions.methods, 'declineShares')
+
+          it('should display accept and decline buttons if share status is pending', () => {
+            const store = createStore({
+              selected: addPropToSelectedFiles(selectedFiles, { status: shareStatus.pending })
+            })
+            wrapper = createShallowMountWrapper({ ...options, store })
+            actionButtons = wrapper.findAll(elSelector.ocButtonStub)
+            acceptButton = wrapper.find(elSelector.acceptButton)
+            declineButton = wrapper.find(elSelector.declineButton)
+
+            expect(actionButtons.length).toEqual(2)
+            expect(acceptButton.exists()).toBeTruthy()
+            expect(acceptButton.text()).toContain('Accept')
+            expect(declineButton.exists()).toBeTruthy()
+            expect(declineButton.text()).toContain('Decline')
+          })
+          describe('when share status is declined', () => {
+            beforeEach(() => {
+              const store = createStore({
+                selected: addPropToSelectedFiles(selectedFiles, { status: shareStatus.declined })
+              })
+              wrapper = createMountWrapper({ ...options, store })
+              actionButtons = wrapper.findAll(elSelector.ocButton)
+              acceptButton = wrapper.find(elSelector.acceptButton)
+              declineButton = wrapper.find(elSelector.declineButton)
+            })
+            it('should display accept button', () => {
+              expect(actionButtons.length).toEqual(1)
+              expect(acceptButton.exists()).toBeTruthy()
+              expect(acceptButton.text()).toContain('Accept')
+              expect(declineButton.exists()).toBeFalsy()
+            })
+            it('should call "acceptShares" when accpet button is clicked', async () => {
```

Comment:
it should not display decline button

Reply:
the function implementation is mocked so it's not possible to check for the changes